src/realm/object-store/sectioned_results.cpp

```diff
@@ -227,25 +281,38 @@ void create_buffered_key(Mixed& key, std::unique_ptr<char[]>& buffer, StringType
     }
 }
 
-ResultsSection::ResultsSection()
-    : m_parent(nullptr)
-    , m_key(Mixed())
+template <typename StringType>
+void create_buffered_key(Mixed& key, std::list<std::string>& buffer, StringType value)
```

Comment:
Inserting into a std::list never invalidates pointers to existing members, which is important as the map's keys are pointing to it and we don't want to have to rebuild the map every time we buffer a new key. It could also be something like `std::vector<std::unique_ptr<char[]>>` but that works out to being mostly the same thing but more complicated to work with.